dotcom-rendering/src/lib/identity.ts

```diff
@@ -22,3 +28,163 @@ export async function isSignedInWithOktaAuthState(): Promise<
 			};
 		});
 }
+
+type OktaAuthState = IdentityAuthState<never, CustomIdTokenClaims>;
+
+type SignedOutWithCookies = { kind: 'SignedOutWithCookies' };
+export type SignedInWithCookies = { kind: 'SignedInWithCookies' };
+type SignedOutWithOkta = { kind: 'SignedOutWithOkta' };
+export type SignedInWithOkta = {
+	kind: 'SignedInWithOkta';
+	accessToken: AccessToken<never>;
+	idToken: IDToken<CustomIdTokenClaims>;
+};
+
+export type AuthStatus =
+	| { kind: 'Pending' }
+	| SignedOutWithCookies
+	| SignedInWithCookies
+	| SignedOutWithOkta
+	| SignedInWithOkta;
+
+// relates to getUserFromCookie() below but not used in user-features - however may be useful for other purposes
+type IdentityUserFromCache = {
+	dates: { accountCreatedDate: string };
+	publicFields: {
+		displayName: string;
+	};
+	statusFields: {
+		userEmailValidated: boolean;
+	};
+	id: number;
+	rawResponse: string;
+} | null;
+
+let userFromCookieCache: IdentityUserFromCache = null;
+
+const decodeBase64 = (str: string): string =>
+	decodeURIComponent(
+		escape(
+			window.atob(
+				str.replace(/-/g, '+').replace(/_/g, '/').replace(/,/g, '='),
+			),
+		),
+	);
+
+const getUserFromCookie = (): IdentityUserFromCache => {
+	if (userFromCookieCache === null) {
+		const cookieData = getCookie({ name: 'GU_U', shouldMemoize: true });
+
+		if (cookieData) {
+			try {
+				const userData = JSON.parse(
+					decodeBase64(cookieData.split('.')[0] ?? ''),
+				) as string[];
+
+				const id = parseInt(userData[0] ?? '', 10);
+				const displayName = decodeURIComponent(userData[2] ?? '');
+				const accountCreatedDate = userData[6];
+				const userEmailValidated = Boolean(userData[7]);
+
+				if (id && accountCreatedDate) {
+					userFromCookieCache = {
+						id,
+						publicFields: {
+							displayName,
+						},
+						dates: { accountCreatedDate },
+						statusFields: {
+							userEmailValidated,
+						},
+						rawResponse: cookieData,
+					};
+				}
+			} catch (e) {
+				// eslint-disable-next-line no-console -- we want to log the error to console, not Sentry
+				console.error('Error parsing GU_U cookie', e);
```

Comment:
Is that true?

Reply:
Perhaps a more apt log `cookie value is malformed`